airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/destination/async/buffers/BufferEnqueue.kt

```diff
@@ -20,6 +22,7 @@ class BufferEnqueue(
     private val memoryManager: GlobalMemoryManager,
     private val buffers: ConcurrentMap<StreamDescriptor, StreamAwareQueue>,
     private val stateManager: GlobalAsyncStateManager,
+    private val streamsWithSuccessStatus: ConcurrentHashSet<StreamDescriptor>,
```

Comment:
this should probably be a map streamdescriptor -> status, so that we can explicitly check whether each stream succeeded or failed (and error if a stream had neither status)